packages/flutter_tools/lib/src/base/utils.dart

```diff
@@ -479,3 +479,19 @@ Match? firstMatchInFile(File file, RegExp regExp) {
   }
   return null;
 }
+
```

Comment:
Logically, they mean the same thing. To capture this idea, I've changed `flavors` to be a set rather than a list.

I am not 100% sure on the hashing function I wrote, but the chance of unexpected collision should be low. We also don't use it anywhere (only the `equals` implementation is used).